// eslint-plugin-ui5/lib/rules/no-for-in.js
/**
 * @fileoverview No usage of for...in loops
 * @author Marc Schleeweiß
 */
"use strict";

//------------------------------------------------------------------------------
// Rule Definition
//------------------------------------------------------------------------------

module.exports = {
    meta: {
        docs: {
            description: "No usage of for...in loops",
            category: "Best Practices",
            recommended: true
        },
        fixable: null,  // or "code" or "whitespace"
        schema: [
            // fill in your schema
        ]
    },

    create: function (context) {

        const ERROR_MSG_NO_FOR_IN = "Unexpected for...in loop"

        //----------------------------------------------------------------------
        // Helpers
        //----------------------------------------------------------------------

        // any helper functions should go here or else delete this section

        //----------------------------------------------------------------------
        // Public
        //----------------------------------------------------------------------

        return {

            ForInStatement(node) {
                context.report(node, ERROR_MSG_NO_FOR_IN);
            }

        };
    }
};
